### PR TITLE
doc: sysbuild: Improve documentation for targets

### LINKED ISSUE #57075: doc: better describe sysbuild targets
**Is your enhancement proposal related to a problem? Please describe.**
It seems that things like `ninja sb_menuconfig` are lacking from the sysbuild documentation

**Describe the solution you'd like**
To have it added

**Describe alternatives you've considered**
None

### PR DESCRIPTION
**Pull request by nordicjm**
Adds information on targets that can be used from sysbuild.

@tejlmand not sure if the terminology here is correct?

Fixes #57075

### CODE REVIEW

File: doc/build/sysbuild/index.rst [hunk 1]
```diff
@@ -386,6 +393,23 @@ specify this file when building with sysbuild, as follows:
...
+ * menuconfig
+ * hardenconfig
+ * guiconfig
+
+For the main application (as is the same without using sysbuild) these can be
+ran normally without any prefix. For other images (including sysbuild), these
+are ran with a prefix of the image name and an underscore e.g. ``sysbuild_`` or
+``mcuboot_``, using ninja or make. Note that flash and debug modes are not
+included when using sysbuild, flashing and debugging should be done using west
+commands instead.
```
Comment (tejlmand) [inline doc/build/sysbuild/index.rst:411]: Sysbuild flashing and debugging using `west` has already been described earlier in this document:
https://docs.zephyrproject.org/latest/build/sysbuild/index.html#sysbuild-flashing-using-west-flash

so I don't think it is needed again to state that `west` must be used for flashing and debugging.

Also the fact that those targets are not part of the list x_config` list above should hint at this. 

But not a blocker for me, so if you really think it should be here, i'm ok with that.

**Review by tejlmand — COMMENTED** [review]
Thanks for this improvement.

File: doc/build/sysbuild/index.rst (hunk 1 above)
Comment (nordicjm) [reply doc/build/sysbuild/index.rst:411]: It's more to present an example of things that are not covered e.g. tfm*, *_report, etc.

Comment (tejlmand) [reply doc/build/sysbuild/index.rst]: > It's more to present an example of things that are not covered e.g. tfm*, *_report, etc.

Thanks for the reasoning.
I think that a subsection describing how such targets can be invoked would make a lot of sense.
Either as part of this PR or a dedicated one. 

For example:
```
Dedicated image build targets
===================

a good description including something like .....
Dedicated image build targets which are not mapped by sysbuild targets, such as `tfm*`, `ram_report`,
`rom_report`, etc can be invoked in the image's build folder, using `ninja` or `west`.

For example when building hello_world using `west`, then rom_report for hello_world can be generated as:
Build hello_world:
west build -b reel_board --sysbuild samples/hello_world

Generating rom_report using `west`:
west build -d build/hello_world -t rom_report

Generating rom_report using `ninja`:
ninja -C build/hello_world rom_report

**Review by tejlmand — DISMISSED** [review]
looks good.

And the intention of describing unmapped build targets makes a lot of sense.

**Review by tejlmand — DISMISSED** [review]
Approved.

PR is getting better between each approval 👍

**Comment by nordicjm** [comment]
Rebased to fix CI failure, please re-approve @tejlmand

**Review by tejlmand — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**